controllers/chaosimpl/utils/record.go

```diff
@@ -46,7 +46,12 @@ type DecodedContainerRecord struct {
 
 func (d *ContianerRecordDecoder) DecodeContainerRecord(ctx context.Context, record *v1alpha1.Record) (decoded DecodedContainerRecord, err error) {
 	var pod v1.Pod
-	podId, containerName := controller.ParseNamespacedNameContainer(record.Id)
+	podId, containerName, err := controller.ParseNamespacedNameContainer(record.Id)
+	if err != nil {
+		// TODO: organize the error in a better way
+		err = NewFailToFindContainer(pod.Namespace, pod.Name, containerName, err)
```

Comment:
If it parses failed, both of the `containerName` and `podId` should be the empty strings, it's non-sense to define this error, I suggest just return it.

Reply:
I did just as you say, but httpchaos threw an error of can't find pod "", so I add this error.

Reply:
But I find that may be the wrong use of ```NewFailToFindContainer``` in httpchaos

Reply:
The `ParseNamespacedNameContainer` returns error should be sth like `too few parts of namespacedname` as [L51](https://github.com/chaos-mesh/chaos-mesh/pull/2141/files#diff-fad6897b0163bb2450778c4bade5d28c847364e881670dd290b3d83b602dd925R51), it's different to `can't find pod ""`.

Reply:
That happended when I haven't add this error.

Reply:
Why? `HTTPChaos` doesn't call `ParseNamespacedNameContainer` now. How could it get an error from the function which it doesn't call?

Reply:
Which line would the httpchaos return that error `can't find pod ""`?

There is nothing magic about `NewFailToFindContainer` method, it just exposed `Error` function to make it works as an error. And the outside world even couldn't check its original fields..

Reply:
This is the e2e-test log of the past without error adding
```
[2021-07-26T04:36:51.037Z] [3] Jul 26 04:36:50.016: FAIL: http chaos doesn't work as expected

[2021-07-26T04:36:51.037Z] [3] Unexpected error:

[2021-07-26T04:36:51.037Z] [3]     <*errors.errorString | 0xc000353f00>: {

[2021-07-26T04:36:51.037Z] [3]         s: "timed out waiting for the condition",

[2021-07-26T04:36:51.037Z] [3]     }

[2021-07-26T04:36:51.037Z] [3]     timed out waiting for the condition

[2021-07-26T04:36:51.037Z] [3] occurred

[2021-07-26T04:36:51.037Z] [3] [JustAfterEach] [HTTPChaos]

[2021-07-26T04:36:51.037Z] [3]   /home/jenkins/agent/workspace/chaos-mesh-e2e/go/src/github.com/chaos-mesh/chaos-mesh/e2e-test/e2e/chaos/basic.go:251

[2021-07-26T04:36:51.037Z] [3] [AfterEach] [Basic]

[2021-07-26T04:36:51.037Z] [3]   /go/pkg/mod/k8s.io/kubernetes@v1.17.2/test/e2e/framework/framework.go:152

[2021-07-26T04:36:51.037Z] [3] STEP: Collecting events from namespace "chaos-mesh-3790".

[2021-07-26T04:36:51.037Z] [3] STEP: Found 25 events.

[2021-07-26T04:36:51.037Z] [3] Jul 26 04:36:50.022: INFO: At 2021-07-26 04:35:45 +0000 UTC - event for http-test: {deployment-controller } ScalingReplicaSet: Scaled up replica set http-test-6c4c75854b to 1

[2021-07-26T04:36:51.037Z] [3] Jul 26 04:36:50.022: INFO: At 2021-07-26 04:35:45 +0000 UTC - event for http-test-6c4c75854b: {replicaset-controller } SuccessfulCreate: Created pod: http-test-6c4c75854b-pfl7j

[2021-07-26T04:36:51.037Z] [3] Jul 26 04:36:50.022: INFO: At 2021-07-26 04:35:45 +0000 UTC - event for http-test-6c4c75854b-pfl7j: {default-scheduler } Scheduled: Successfully assigned chaos-mesh-3790/http-test-6c4c75854b-pfl7j to chaos-mesh-worker2

[2021-07-26T04:36:51.037Z] [3] Jul 26 04:36:50.022: INFO: At 2021-07-26 04:35:46 +0000 UTC - event for http-test-6c4c75854b-pfl7j: {kubelet chaos-mesh-worker2} Started: Started container http

[2021-07-26T04:36:51.037Z] [3] Jul 26 04:36:50.022: INFO: At 2021-07-26 04:35:46 +0000 UTC - event for http-test-6c4c75854b-pfl7j: {kubelet chaos-mesh-worker2} Created: Created container http

[2021-07-26T04:36:51.037Z] [3] Jul 26 04:36:50.022: INFO: At 2021-07-26 04:35:46 +0000 UTC - event for http-test-6c4c75854b-pfl7j: {kubelet chaos-mesh-worker2} Pulled: Container image "localhost:5000/pingcap/e2e-helper:latest" already present on machine

[2021-07-26T04:36:51.037Z] [3] Jul 26 04:36:50.022: INFO: At 2021-07-26 04:35:49 +0000 UTC - event for http-chaos: {desiredphase } Started: Experiment has started

[2021-07-26T04:36:51.037Z] [3] Jul 26 04:36:50.022: INFO: At 2021-07-26 04:35:49 +0000 UTC - event for http-chaos: {finalizer } FinalizerInited: Finalizer has been inited

[2021-07-26T04:36:51.037Z] [3] Jul 26 04:36:50.022: INFO: At 2021-07-26 04:35:50 +0000 UTC - event for http-chaos: {records } Failed: Failed to apply chaos: Pod "" not found

[2021-07-26T04:36:51.037Z] [3] Jul 26 04:36:50.022: INFO: At 2021-07-26 04:35:50 +0000 UTC - event for http-chaos: {records } Failed: Failed to apply chaos: Pod "" not found

[2021-07-26T04:36:51.037Z] [3] Jul 26 04:36:50.022: INFO: At 2021-07-26 04:35:50 +0000 UTC - event for http-chaos: {records } Failed: Failed to apply chaos: Pod "" not found

[2021-07-26T04:36:51.038Z] [3] Jul 26 04:36:50.022: INFO: At 2021-07-26 04:35:50 +0000 UTC - event for http-chaos: {records } Failed: Failed to apply chaos: Pod "" not found

[2021-07-26T04:36:51.038Z] [3] Jul 26 04:36:50.022: INFO: At 2021-07-26 04:35:50 +0000 UTC - event for http-chaos: {records } Failed: Failed to apply chaos: Pod "" not found

[2021-07-26T04:36:51.038Z] [3] Jul 26 04:36:50.022: INFO: At 2021-07-26 04:35:50 +0000 UTC - event for http-chaos: {desiredphase } Updated: Successfully update desiredPhase of resource

[2021-07-26T04:36:51.038Z] [3] Jul 26 04:36:50.022: INFO: At 2021-07-26 04:35:50 +0000 UTC - event for http-chaos: {finalizer } Updated: Successfully update finalizer of resource

[2021-07-26T04:36:51.038Z] [3] Jul 26 04:36:50.022: INFO: At 2021-07-26 04:35:50 +0000 UTC - event for http-chaos: {records } Updated: Successfully update records of resource

[2021-07-26T04:36:51.038Z] [3] Jul 26 04:36:50.022: INFO: At 2021-07-26 04:35:50 +0000 UTC - event for http-chaos: {records } Failed: Failed to apply chaos: Pod "" not found

[2021-07-26T04:36:51.038Z] [3] Jul 26 04:36:50.022: INFO: At 2021-07-26 04:35:50 +0000 UTC - event for http-chaos: {records } Failed: Failed to apply chaos: Pod "" not found

[2021-07-26T04:36:51.038Z] [3] Jul 26 04:36:50.022: INFO: At 2021-07-26 04:35:50 +0000 UTC - event for http-chaos: {records } Failed: Failed to apply chaos: Pod "" not found

[2021-07-26T04:36:51.038Z] [3] Jul 26 04:36:50.022: INFO: At 2021-07-26 04:35:51 +0000 UTC - event for http-chaos: {records } Failed: Failed to apply chaos: Pod "" not found

[2021-07-26T04:36:51.038Z] [3] Jul 26 04:36:50.022: INFO: At 2021-07-26 04:35:52 +0000 UTC - event for http-chaos: {records } Failed: Failed to apply chaos: Pod "" not found

[2021-07-26T04:36:51.038Z] [3] Jul 26 04:36:50.022: INFO: At 2021-07-26 04:35:55 +0000 UTC - event for http-chaos: {records } Failed: Failed to apply chaos: Pod "" not found

[2021-07-26T04:36:51.038Z] [3] Jul 26 04:36:50.022: INFO: At 2021-07-26 04:36:00 +0000 UTC - event for http-chaos: {records } Failed: Failed to apply chaos: Pod "" not found

[2021-07-26T04:36:51.038Z] [3] Jul 26 04:36:50.022: INFO: At 2021-07-26 04:36:10 +0000 UTC - event for http-chaos: {records } Failed: Failed to apply chaos: Pod "" not found

[2021-07-26T04:36:51.038Z] [3] Jul 26 04:36:50.022: INFO: At 2021-07-26 04:36:31 +0000 UTC - event for http-chaos: {records } Failed: Failed to apply chaos: Pod "" not found
```

Reply:
There are many commits after that time e2e test, how about trigger a new CI test, and let's see what gonna happen.

Reply:
Besides, we could see the latest e2e test is passed, so I think it's not a problem for you mentioned case.

Reply:
Which version of code do you referring to?  The one without error throwing?

Reply:
I mean we could just return the error as I suggested, and let's see the ci test result. If there are still some errors, I would help you dive into the root cause.

Reply:
It's still an issue.